test/e2e_node/hugepages_test.go

```diff
@@ -122,24 +170,39 @@ func configureHugePages() error {
 		return err
 	}
 	e2elog.Logf("HugePages_Total is set to %v", numHugePages)
-	if numHugePages == 50 {
+	if int64(numHugePages) == pageCount {
 		return nil
 	}
-	return fmt.Errorf("expected hugepages %v, but found %v", 50, numHugePages)
+	return fmt.Errorf("expected hugepages %v, but found %v", pageCount, numHugePages)
 }
 
 // releaseHugePages releases all pre-allocated hugepages
 func releaseHugePages() error {
 	return exec.Command("/bin/sh", "-c", "echo 0 > /proc/sys/vm/nr_hugepages").Run()
 }
 
-// isHugePageSupported returns true if the default hugepagesize on host is 2Mi (i.e. 2048 kB)
-func isHugePageSupported() bool {
+// getDefaultHugePageSize returns the default huge page size, and a boolean if huge pages are supported
+func getDefaultHugePageSize() (resource.Quantity, bool) {
```

Comment:
I wonder if instead of `bool`, this could use the `err` type to indicate whether huge pages are supported? That feels slightly more like idiomatic Go to me.

Reply:
Hmm, fair point. I think`err` make sense since it is called `getDefaultHugePageSize` not `isSupported` etc.

The `res, ok := check()` is a commonly used thing in go tho, e.g. https://golang.org/ref/spec#Receive_operator

Reply:
Fair point - I guess that is how dictionary lookup works as well... Tbh I don't feel super strong. `err` feels slightly more idiomatic, but I defer to your judgement :)

Reply:
Ill rewrite to `err` when i get time ;)